world-explore/src/app/home/home.page.ts

```diff
@@ -84,16 +88,20 @@ export class HomePage implements OnInit {
       });
     }
   }
-  
+
   getKeys(obj: any): string[] {
     return Object.keys(obj).sort();
   }
 
   filterCountries() {
-    this.filteredCountries = this.countries.filter(country => 
-      country.country.toLowerCase().includes(this.searchTerm.toLowerCase()) &&
-      (!this.showFavorites || this.favoriteCountries.has(country.country))
-    );
+    const lowerCaseSearchTerm = this.searchTerm.toLowerCase();
+
+    this.filteredCountries = this.countries.filter(country => {
```

Comment:
1- Combines the filtering logic into a single return statement for clarity and readability.
2- Logical Condition:
Returns true for countries that match the search term (matchesSearchTerm) and either !this.showFavorites (if showFavorites is false, ignore the favorite filter) or isFavorite (if showFavorites is true, include only favorite countries).
3- Separate Variables:
1-matchesSearchTerm: Checks if lowerCaseCountryName includes lowerCaseSearchTerm, ensuring case-insensitive matching.
2- isFavorite: Determines if country.country is present in this.favoriteCountries.

Reply:
ok, more readable, thanks